### PR TITLE
Log query

### PR DESCRIPTION
**Pull request by bquenin**

### CODE REVIEW

File: api/rpc/logs/logs.proto [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+package logs;
+
+service Logs {
+    rpc Get (GetRequest) returns (GetReply) {
+    }
+}
+
+message LogEntry {
+    string timestamp = 1;
+    string TimeId = 2;
```
Comment (subfuzion) [inline api/rpc/logs/logs.proto]: Please conform to the [protocol buffers style guide](https://developers.google.com/protocol-buffers/docs/style) for naming fields. Use all lowercase and underscores. The language-specific drivers will convert them to to accessors that conform to language (Go) conventions.

File: api/server/config.go [hunk 2]
```diff
@@ -8,9 +8,10 @@ import (
 type Config struct {
 	Port          string
 	EtcdEndpoints []string
+	esURL         string
```
Comment (subfuzion) [inline api/server/config.go]: I think it's ok to expose this as a public field, and the field name might as well be a bit more descriptive. Maybe `ElasticsearchURL`?

File: cmd/amplifier/main.go [hunk 3]
```diff
@@ -27,12 +28,14 @@ var (
 	config        server.Config
 	port          string
 	etcdEndpoints string
+	esURL         string
```
Comment (subfuzion) [inline cmd/amplifier/main.go]: Nitpicking here, but I think I would prefer to see `elasticsearchURL`...

File: cmd/amplifier/main.go [hunk 4]
```diff
@@ -27,12 +28,14 @@ var (
...
 )
 
 func parseFlags() {
 	// set up flags
 	flag.StringVarP(&port, "port", "p", defaultPort, "server port (default '"+defaultPort+"')")
 	flag.StringVarP(&etcdEndpoints, "endpoints", "e", etcdDefaultEndpoints, "etcd comma-separated endpoints")
+	flag.StringVarP(&esURL, "esURL", "s", esDefaultURL, "elasitc serach url (default '"+esDefaultURL+"')")
```
Comment (subfuzion) [inline cmd/amplifier/main.go]: Command line flags should be all lowercase:  `esURL` -> `elasticsearch`.

File: data/elasticsearch/elasticsearch.go [hunk 5]
```diff
@@ -86,3 +83,8 @@ func (es *ElasticSearch) All(esIndex string) []*elastic.SearchHit {
 	}
 	return searchResult.Hits.Hits
 }
+
+// GetNative returns the native elastic search client
+func (es *ElasticSearch) GetNative() *elastic.Client {
```
Comment (subfuzion) [inline data/elasticsearch/elasticsearch.go]: Will any other consumers actually need access to the internal client? Subjective, but if so, consider renaming to `GetClient`.

**Comment by subfuzion** [comment]
@bertrand-quenin: looking pretty good. Mostly just a few nitpicky comments.

File: api/server/config.go [hunk 6]
```diff
@@ -6,11 +6,12 @@ import (
 
 // Config is used for amplifier configuration settings
 type Config struct {
-	Port          string
-	EtcdEndpoints []string
+	Port             string
+	EtcdEndpoints    []string
+	ElasticSearchURL string
```
Comment (subfuzion) [inline api/server/config.go]: Really hate to nitpick like this, but technically it is `Elasticsearch` not `ElasticSearch`. https://www.elastic.co/products/elasticsearch

File: api/server/server.go [hunk 7]
```diff
@@ -43,3 +50,14 @@ func initEtcd(config Config) {
 	}
 	log.Printf("connected to etcd at %v", strings.Join(Store.Endpoints(), ","))
 }
+
+// fail fast on initialization errors; there's no point in attempting
+// to continue in a degraded state if there are problems at start up
+func initElasticSearch(config Config) {
```
Comment (subfuzion) [inline api/server/server.go]: -> `initElasticsearch`

File: cmd/amplifier/main.go [hunk 8]
```diff
@@ -24,15 +25,17 @@ var (
 
 // config vars - used for generating a config from command line flags
 var (
-	config        server.Config
-	port          string
-	etcdEndpoints string
+	config           server.Config
+	port             string
+	etcdEndpoints    string
+	elasticSearchURL string
```
Comment (subfuzion) [inline cmd/amplifier/main.go]: -> `elasticsearchURL`

File: cmd/amplifier/main.go [hunk 9]
```diff
@@ -9,8 +9,9 @@ import (
 )
 
 const (
-	defaultPort          = ":50101"
-	etcdDefaultEndpoints = "http://localhost:2379"
+	defaultPort             = ":50101"
+	etcdDefaultEndpoints    = "http://localhost:2379"
+	elasticSearchDefaultURL = "http://localhost:9200"
```
Comment (subfuzion) [inline cmd/amplifier/main.go]: -> elasticsearchDefaultURL`

File: data/etcd/store_test.go [hunk 10]
```diff
@@ -12,33 +12,40 @@ import (
...
 var (
-	config        server.Config
-	store         data.Store
-	port          string
-	etcdEndpoints string
+	config           server.Config
+	store            data.Store
+	port             string
+	etcdEndpoints    string
+	elasticSearchURL string
```
Comment (subfuzion) [inline data/etcd/store_test.go]: -> `elasticsearchURL`

**Comment by subfuzion** [comment]
@bertrand-quenin PRs should have a reference to an issue and some kind of description that summarizes what to expect out of the PR. Here are examples: https://github.com/appcelerator/amp/pull/7, https://github.com/appcelerator/amp/pull/17, https://github.com/appcelerator/amp/pull/6. Thanks.

**Comment by chrisccoy** [comment]
For me, the test is failing. What is the dependency? I am running `./swarm start min` mode

=== RUN   TestShouldGetAHundredLogEntries
--- FAIL: TestShouldGetAHundredLogEntries (0.04s)
    log_test.go:74: expected: 100, got: 0
FAIL
exit status 1

File: Makefile [hunk 11]
```diff
@@ -44,8 +44,8 @@ clean:
 	@rm -f $$(which amp)
 
 test:
-	@go test $(REPO)/api/server
-	@go test $(REPO)/data/etcd
+	@go test -v $(REPO)/api/server
```
Comment (chrisccoy) [inline Makefile]: Do we want verbose mode? It looks kind of ugly. Redundant lines in output

File: data/elasticsearch/elasticsearch.go [hunk 12]
```diff
@@ -5,30 +5,27 @@ import (
...
-	if err != nil {
-		// TODO: Handle error
-		panic(err)
-	}
+	return err
 }
 
 // CreateIndexIfNotExists Creates an index if it doesn't already exists
-func (es *ElasticSearch) CreateIndexIfNotExists(esIndex string, esType string, mapping string) {
+func (es *Elasticsearch) CreateIndexIfNotExists(esIndex string, esType string, mapping string) {
```
Comment (chrisccoy) [inline data/elasticsearch/elasticsearch.go]: It might be best to simply return the error for now instead of panic?

**Comment by subfuzion** [comment]
I also get a test faillure although the error is different.

```
$ make test
2016/08/10 10:36:18 connecting to etcd at http://localhost:2379
2016/08/10 10:36:18 connected to etcd at http://localhost:2379
2016/08/10 10:36:18 connecting to elasticsearch at http://localhost:9200
=== RUN   TestShouldGetAHundredLogEntries
--- FAIL: TestShouldGetAHundredLogEntries (0.00s)
        log_test.go:71: could not get logs: rpc error: code = 14 desc = grpc: RPC failed fast due to transport failure
FAIL
exit status 1
FAIL    github.com/appcelerator/amp/api/server  1.021s
make: *** [test] Error 1
```

**Comment by subfuzion** [comment]
@bertrand-quenin As is, this will never pass the CI build regardless. Please update `docker-compose.test.yml` to add the necessary services. The `sut` command should just call `make test`. You will need to make sure that `sut` supplies the environment variables to `make` to use the service names (not localhost).

**Comment by subfuzion** [comment]
@bertrand-quenin My bad for providing the example code with service and project implementation files under `api/server`, but they -- along with your log service implementation and test -- should be in the appropriate service directories. In your case, that means moving `log.go` and `log_test.go` to the `logs` directory (which I'd like renamed to just `log`). It's fine to keep implementation code in the same directory as the proto definition and generated stub code / interface.

**Pull request closed without merging**